new-components/reporters/elasticsearch/cmd/main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/go-errors/errors"
+
+	"github.com/smithy-security/smithy/sdk/component"
+
+	"github.com/smithy-security/smithy/new-components/reporters/elasticsearch/internal/reporter"
+)
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	if err := Main(ctx); err != nil {
+		log.Fatalf("unexpected error: %v", err)
+	}
+}
+
+func Main(ctx context.Context, opts ...component.RunnerOption) error {
+	conf, err := reporter.NewConf(nil)
+	if err != nil {
+		return errors.Errorf("could not create new configuration: %w", err)
+	}
+	opts = append(opts, component.RunnerWithComponentName("elasticsearch"))
+
+	client, err := reporter.GetESClient(conf)
+	if err != nil {
+		return errors.Errorf("could not create elasticsearch client: %w", err)
+	}
+	withClient := reporter.ElasticsearchReporterWithClient(client)
```

Comment:
You can avoid this if you pass the options in the constructor

This is a mix between builder pattern and functional options 🤔 

https://www.bytesizego.com/blog/10-years-functional-options-golang